low-level/sensors/impl/generic/cpu_fault_sensor.py

```diff
@@ -180,8 +183,7 @@ def run(self):
                 to_update = True
 
         except Exception as e:
-            logger.error(f"Error while processing cpu info, shutting down CPUFaultSensor : {e}")
-            self.shutdown()
+            raise Exception(f"Failed while processing cpu info, {e}")
```

Comment:
For exception in run() method, raising `module_thread:ThreadException()` might be better.

Reply:
In initialize() method previously we also used to return True and False and take action according to the return status. Are we just raising an exception now or returning False. Both are possible, right?

Reply:
`For exception in run() method, raising module_thread:ThreadException() might be better.`
Sensor itself a thread, so keeping that for run() alone won't suite.

`In initialize() method previously we used to return True and False and take action accordingly. Are we just raising an exception now or returning False. Both are possible, right?` Same as previous still. In case exception raised during initailize(), recovery will retry initialize().

Reply:
in `check_and_conclude_initialization()` method the `SensorThreadState.WAITING` state has become dead now I think, since while calling initialize() if there is an exception it will pass that exception to sspl_ll_d and try execution again until retry count is exceeded. If after those any retries it still fails we are logging the error again, so the state will never go to WAITING now. probably we can remove that code.

Reply:
Even in `thread_controller.py:run()` method we are checking WAITING state and delaying `"SSPL-LL service has started successfully."` msg is all threads are not started successfully. But now exception will occur instead of False return value so what is behaviour observed in this case?

Reply:
Got it.
Can you review module_thread.py? there I set the status as FAILED. So all thread status are known now, it will allow SSPL to start. @sumedhak27

Reply:
Yes if there is no need for Thread WAITING state, we can avoid that code.

Reply:
Done.